### ISSUE TITLE
Is there a way to treat all errors as warnings?

### ISSUE THREAD

**Issue by adamralph** [DotNetAnalyzers/StyleCopAnalyzers#2199]
Similar to the old StyleCop

```xml
<StyleCopTreatErrorsAsWarnings>true</StyleCopTreatErrorsAsWarnings>
```

(I tried that and it had no effect.)

**Comment by sharwell** [comment]
Hi @adamralph,

This project doesn't report any errors at this time. However, there are two ways I know where the use of this project could result in build errors:

1. A rule set file is in use which overrides the default severity for one or more StyleCop Analyzers diagnostic IDs (from warning &rarr; error)
2. The project is using "Treat Warnings as Errors"

The solution to the first one is to simply update the rule set file to go back to warnings (the default). The solution to the second is a bit more involved:

1. Set Treat Warnings as Errors back to false
2. Open the rule set file for your project (or create one if you aren't already using one)
3. Update the severity of built-in warnings from Warning &rarr; Error, while leaving StyleCop Analyzers diagnostics still set at Warning.

    :bulb: If you were using Treat Warnings as Errors, the specific warnings you likely want to set to Error severity are the ones located in the **Microsoft.CodeAnalysis.CSharp** group in the rule set editor (these are the ones with IDs of the form CS####).

**Comment by adamralph** [comment]
Thanks @sharwell.

I am indeed using "Treat Warnings as Errors" = true so it, indeed it makes sense that warnings from StyleCop.Analyzers are showing as errors.

I've decided to simply set this to false for now and see how that goes.

**Comment by thomaslevesque** [comment]
> 💡 If you were using Treat Warnings as Errors, the specific warnings you likely want to set to Error severity are the ones located in the Microsoft.CodeAnalysis.CSharp group in the rule set editor (these are the ones with IDs of the form CS####).

Unfortunately, this group doesn't contain all possible C# warnings, and also contains rules that are not warnings (Info or Hidden severity). Is there a full list of C# warnings somewhere?
